src/olympia/promoted/tasks.py

```diff
@@ -106,3 +110,66 @@ def on_stripe_charge_failed(event):
         from_email=settings.ADDONS_EMAIL,
         recipient_list=["amo-admins@mozilla.com"],
     )
+
+
+@task
+@use_primary_db
+def on_stripe_customer_subscription_deleted(event):
+    event_id = event.get("id")
+    event_type = event.get("type")
+
+    if event_type != "customer.subscription.deleted":
+        log.error(
+            'invalid event "%s" received (event_id=%s).', event_type, event_id
+        )
+        return
+
+    # This event should contain a `subscription` object.
+    subscription = event.get("data", {}).get("object")
+    if not subscription:
+        log.error("no subscription object in event (event_id=%s).", event_id)
+        return
+
+    subscription_id = subscription["id"]
+    log.info(
+        'received "%s" event with subscription_id=%s.',
+        event_type,
+        subscription_id,
+    )
+
+    try:
+        sub = PromotedSubscription.objects.get(
+            stripe_subscription_id=subscription_id
+        )
+    except PromotedSubscription.DoesNotExist:
+        log.exception(
+            'received a "%s" event (event_id=%s) for a non-existent'
+            " promoted subscription (subscription_id=%s).",
+            event_type,
+            event_id,
+            subscription_id,
+        )
+        return
+
+    if sub.cancelled_at:
+        log.info(
+            "PromotedSubscription %s already cancelled, ignoring event"
+            " (event_id=%s).",
```

Comment:
```suggestion
            "PromotedSubscription %s already cancelled, ignoring event "
            "(event_id=%s).",
```
assuming Black doesn't also mis-format strings over multiple lines 😛